apps/contracts/project.json

```diff
@@ -3,6 +3,18 @@
     "sourceRoot": "apps/contracts",
     "projectType": "application",
     "targets": {
+        "serve": {
+            "executor": "nx:run-commands",
+            "options": {
+                "commands": [
+                    "hardhat compile",
+                    "hardhat deploy",
```

Comment:
Are they executed in order? `hardhat node` should be before `hardhat deploy`

Reply:
Mmh `hardhat deploy` uses a temporary Hardhat network maybe? I feel like the right command should be `hardhat deploy --network localhost`. Could you check?

Reply:
You are right. `hardhat deploy` is deploying to a network named `hardhat` and not `localhost`. I was wrong in the previous comment. It probably worked for me since I had deployed the same contract in `localhost` as well previously.

I have updated the commands now. We are running two commands in parallel, one to start node and another to compile+deploy. Its working well now, but if we notice the deploy is getting ran before the node start, we can add a `sleep` before compile.